### PR TITLE
Fix `agg_numerical_column` function

### PR DESCRIPTION
**Pull request by ThaminduR**
Logical error in the `agg_numerical_column` has been fixed.

Related Issue - https://github.com/ThaminduR/spark-privacy-preserver/issues/13

### CODE REVIEW

**Comment by agessner** [comment]
Hey @ThaminduR thanks for fixing it! I don't know much about k-anonymization, but here are the tests @Marcaoandradenogueira and I have made. They are giving us the following results:

### input
| column1 | column2 | column3 | column4 | column5 |
| - | - | - | - | - |
| 600000 | 1 | test1 | x | 20 |
| 600000 | 1 | test1 | y | 30 |
| 800000 | 2 | test2 | x | 50 |
| 800000 | 2 | test3 | x | 45 |
| 800000 | 1 | test2 | y | 35 |
| 400000 | 2 | test3 | y | 20 |

### params

```python
categorical = set(('column2', 'column3', 'column4'))
sensitive_column = 'column4'
feature_columns = ['column1', 'column2', 'column3']
k = 2
```

### k_anonymize

| column1 | column2 | column3 | column4 | count |
| - | - | - | - | - |
| 400000-600005 | 1, 2 | test3, test1 | x | 1 |
| 400000-600005 | 1, 2 | test3, test1 | y | 2 |
| 800000 | 1, 2 | test3, test2 | x | 2 |
| 800000 | 1, 2 | test3, test2 | y | 1 |

### k_anonymize_w_user

| column1 | column2 | column3 | column4 | column5 |
| - | - | - | - | - |
| 05-10 | 1, 2 | test1, test3 | x | 20 |
| 05-10 | 1, 2 | test1, test3 | y | 30 |
| 05-10 | 1, 2 | test1, test3 | x | 20 |
| 800000 | 2, 1 | test2, test3 | x | 50 |
| 800000 | 2, 1 | test2, test3 | y | 45 |
| 800000 | 2, 1 | test2, test3 | y | 35 |

The k_anonymize appears to be fixed, but the k_anonymize_w_user is having the same problem as before. Maybe it's some parameter we are passing wrong or the data is not a good fit. It just felt weird having this 05-10 result.

Thanks!

**Comment by ThaminduR** [comment]
97c2211 should fix the issue with `k_anonymize_w_user` function. 
Also ,now the numerical aggregating function will handle 0 or 5 as the last digit case, gracefully.

File: spark_privacy_preserver/mondrian_utils/utility.py [hunk 1]
```diff
@@ -183,23 +183,29 @@ def agg_numerical_column(series):
             else:
                 string = '0-'
         else:
-            min_start = minm[:-1]
-            if(minimum >= int(min_start+'5')):
-                string = min_start+'5-'
+            if (minm[-1]=='0'):
+            string = minm +"-"
```
Comment (agessner) [inline spark_privacy_preserver/mondrian_utils/utility.py]: I think you got the wrong indentation level here :)

File: spark_privacy_preserver/mondrian_utils/utility.py [hunk 2]
```diff
@@ -183,23 +183,29 @@ def agg_numerical_column(series):
...
             if(maximum >= 5):
                 string += "10"
             else:
                 string += '5'
         else:
-            max_start = maxm[:-1]
-            if(maximum >= int(max_start+'5')):
-                string += str(int(max_start+'0') + 10)
+            if(maxm[-1]=='0'):
+            string += maxm
```
Comment (agessner) [inline spark_privacy_preserver/mondrian_utils/utility.py]: here too

File: spark_privacy_preserver/mondrian_utils/utility.py (hunk 1 above)
Comment (ThaminduR) [reply spark_privacy_preserver/mondrian_utils/utility.py]: Fixed in 5c2760a

**Comment by ThaminduR** [comment]
if the test cases are giving the correct output, then I'll merge this.

**Pull request merged**